OpenRA.Mods.Common/Traits/World/JumpjetActorLayer.cs

```diff
@@ -72,28 +72,31 @@ WPos ICustomMovementLayer.CenterOfCell(CPos cell)
 			return pos + new WVec(0, 0, height[cell] - pos.Z);
 		}
 
-		bool ValidTransitionCell(CPos cell, MobileInfo mi)
+		bool ValidTransitionCell(CPos cell, LocomotorInfo li)
 		{
 			var terrainType = map.GetTerrainInfo(cell).Type;
-			if (!mi.JumpjetTransitionTerrainTypes.Contains(terrainType) && mi.JumpjetTransitionTerrainTypes.Any())
+			var jli = li as JumpjetLocomotorInfo;
+			if (!jli.JumpjetTransitionTerrainTypes.Contains(terrainType) && jli.JumpjetTransitionTerrainTypes.Any())
```

Comment:
Minor point that occurs a few times: `as` followed with immediate dereferencing is better served with a normal cast (i.e. `var jli = (JumpjetLocomotorInfo)li;`).

If you're wrong about the type and use a cast, you get an informative InvalidCastException at runtime.
If you're wrong about the type and use `as`, you get a NullReferenceException at runtime, and sadness when we have try and work out exactly what might have been null when we get the stack trace.